### PR TITLE
Conversion of buttons to pf4 button compontent (Group 1)

### PR DESCRIPTION
**Pull request by sg00dwin**
Includes:
- Buttons that are not within modals 
  - (excluding remove and reorder icon buttons)
  - (excluding "edit yaml" link buttons)
- "Add" field inline buttons

### CODE REVIEW

File: frontend/public/components/cluster-settings/basicauth-idp-form.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { ButtonBar, PromiseComponent, history, AsyncComponent } from '../utils';
 import { addIDP, getOAuthResource, redirectToOAuthPage } from './';
 import { IDPNameInput } from './idp-name-input';
 import { IDPCAFileInput } from './idp-cafile-input';
+import { ActionGroup, Button } from '@patternfly/react-core';
```
Comment (rhamilto) [inline frontend/public/components/cluster-settings/basicauth-idp-form.tsx]: Here and the other files:  we put external dependency imports in the first group (move this import to line 4).

File: frontend/public/components/modals/taints-modal.tsx [hunk 2]
```diff
@@ -4,6 +4,8 @@ import * as React from 'react';
 import { Dropdown, EmptyBox, PromiseComponent } from '../utils';
 import { K8sKind, k8sPatch, NodeKind, Taint } from '../../module/k8s';
 import { createModalLauncher, ModalBody, ModalComponentProps, ModalSubmitFooter, ModalTitle } from '../factory';
+import { Button, } from '@patternfly/react-core';
```
Comment (rhamilto) [inline frontend/public/components/modals/taints-modal.tsx]: Remove trailing comma

File: frontend/public/components/modals/tolerations-modal.tsx [hunk 3]
```diff
@@ -4,6 +4,8 @@ import * as React from 'react';
 import { Dropdown, EmptyBox, PromiseComponent } from '../utils';
 import { K8sKind, k8sPatch, Toleration, TolerationOperator } from '../../module/k8s';
 import { createModalLauncher, ModalBody, ModalComponentProps, ModalSubmitFooter, ModalTitle } from '../factory';
+import { Button, } from '@patternfly/react-core';
```
Comment (rhamilto) [inline frontend/public/components/modals/tolerations-modal.tsx]: Remove trailing comma

File: frontend/public/components/operator-lifecycle-manager/create-operand.tsx [hunk 4]
```diff
@@ -141,8 +140,20 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = (props) => {
...
+        <ActionGroup className="pf-c-form">
+          <Button
+            onClick={submit}
+            type="submit"
+            variant="primary">
+            Create
+          </Button>
+          <Button
+            component="a"
+            href={`/k8s/ns/${props.namespace}/clusterserviceversions/${props.clusterServiceVersion.metadata.name}/${referenceForModel(props.operandModel)}`} id="cancel"
```
Comment (rhamilto) [inline frontend/public/components/operator-lifecycle-manager/create-operand.tsx]: `id="cancel"` is new.  Do you need it?  If so, it should go on its own line.

File: frontend/public/components/secrets/create-secret.tsx [hunk 5]
```diff
@@ -854,19 +865,26 @@ class GenericSecretForm extends React.Component<GenericSecretFormProps, GenericS
     const secretEntriesList = _.map(this.state.secretEntriesArray, (entryData, index) => {
       return <div className="co-create-secret__form-entry-wrapper" key={entryData.uid}>
         {_.size(this.state.secretEntriesArray) > 1 && <div className="co-create-secret-form__link--remove-entry">
-          <button className="btn btn-link" type="button" onClick={() => this.removeEntry(index)}>
+          <Button
+            type="button"
+            onClick={() => this.removeEntry(index)}
+            variant="link">
             <i className="fa fa-minus-circle" aria-hidden="true" /> Remove Key/Value
```
Comment (rhamilto) [inline frontend/public/components/secrets/create-secret.tsx]: Update icon

File: frontend/public/components/secrets/create-secret.tsx [hunk 6]
```diff
@@ -854,19 +865,26 @@ class GenericSecretForm extends React.Component<GenericSecretFormProps, GenericS
...
     return (
       <React.Fragment>
         {secretEntriesList}
-        <button className="btn btn-link co-create-secret-form__link--add-entry" type="button" onClick={() => this.addEntry()}>
+        <Button
+          className="co-create-secret-form__link--add-entry pf-m-link--align-left"
+          onClick={() => this.addEntry()}
+          type="button"
+          variant="link">
           <i className="fa fa-plus-circle" aria-hidden="true" /> Add Key/Value
```
Comment (rhamilto) [inline frontend/public/components/secrets/create-secret.tsx]: Update icon

File: frontend/public/components/utils/download-button.tsx [hunk 7]
```diff
@@ -23,9 +23,9 @@ export const DownloadButton: React.FC<DownloadButtonProps> = (props) => {
   };
 
   return <React.Fragment>
-    <button className="btn btn-primary" style={{marginBottom: 10}} disabled={inFlight} type="button" onClick={() => download()}>
+    <Button variant="primary" style={{marginBottom: 10}} disabled={inFlight} type="button" onClick={() => download()}>
       <i className="fa fa-fw fa-download" aria-hidden="true" />&nbsp;Download{inFlight && <React.Fragment>ing...</React.Fragment>}
```
Comment (rhamilto) [inline frontend/public/components/utils/download-button.tsx]: Update icon

File: frontend/public/components/utils/download-button.tsx [hunk 8]
```diff
@@ -23,9 +23,9 @@ export const DownloadButton: React.FC<DownloadButtonProps> = (props) => {
   };
 
   return <React.Fragment>
-    <button className="btn btn-primary" style={{marginBottom: 10}} disabled={inFlight} type="button" onClick={() => download()}>
+    <Button variant="primary" style={{marginBottom: 10}} disabled={inFlight} type="button" onClick={() => download()}>
```
Comment (rhamilto) [inline frontend/public/components/utils/download-button.tsx]: Props
```suggestion
    <Button variant="primary" style={{marginBottom: 10}} isDisabled={inFlight} type="button" onClick={() => download()}>
```

File: frontend/public/components/utils/list-input.tsx [hunk 9]
```diff
@@ -1,6 +1,8 @@
 import * as React from 'react';
 import * as _ from 'lodash-es';
 import * as classNames from 'classnames';
+import { Button, } from '@patternfly/react-core';
```
Comment (rhamilto) [inline frontend/public/components/utils/list-input.tsx]: Remove trailing comma

File: frontend/public/components/edit-yaml.jsx [hunk 10]
```diff
@@ -364,19 +364,21 @@ export const EditYAML = connect(stateToProps)(
...
-                      <button className="btn btn-default" id="cancel" onClick={() => this.onCancel()}>Cancel</button>
-                      {download && <button type="submit" className="btn btn-default pull-right hidden-sm hidden-xs" onClick={() => this.download()}><i className="fa fa-download"></i>&nbsp;Download</button>}
+                      <ActionGroup className="pf-c-form__group--no-top-margin">
+                        {create && <Button type="submit" variant="primary" id="save-changes" onClick={() => this.save()}>Create</Button>}
+                        {!create && !readOnly && <Button type="submit" variant="primary" id="save-changes" onClick={() => this.save()}>Save</Button>}
+                        {!create && <Button type="submit" variant="secondary" id="reload-object" onClick={() => this.reload()}>Reload</Button>}
+                        <Button variant="secondary" id="cancel" onClick={() => this.onCancel()}>Cancel</Button>
+                        {download && <Button type="submit" variant="secondary" className="pf-c-button--align-right hidden-sm hidden-xs" onClick={() => this.download()}><i className="fa fa-download"></i>&nbsp;Download</Button>}
```
Comment (rhamilto) [inline frontend/public/components/edit-yaml.jsx]: Update icon

File: frontend/public/components/routes/create-route.tsx [hunk 11]
```diff
@@ -355,11 +355,21 @@ export class CreateRoute extends React.Component<null, CreateRouteState> {
...
           </div> }
           <ButtonBar errorMessage={this.state.error} inProgress={this.state.inProgress}>
-            <button type="submit"
-              disabled={(!this.state.name || !this.state.service || !this.state.targetPort || (this.state.secure && !this.state.termination))}
-              className="btn btn-primary"
-              id="save-changes">Create</button>
-            <Link to={formatNamespacedRouteForResource('routes')} className="btn btn-default" id="cancel">Cancel</Link>
+            <ActionGroup className="pf-c-form">
+              <Button type="submit"
+                disabled={(!this.state.name || !this.state.service || !this.state.targetPort || (this.state.secure && !this.state.termination))}
```
Comment (rhamilto) [inline frontend/public/components/routes/create-route.tsx]:
```suggestion
                isDisabled={(!this.state.name || !this.state.service || !this.state.targetPort || (this.state.secure && !this.state.termination))}
```

File: frontend/public/components/environment.jsx [hunk 12]
```diff
@@ -508,13 +508,15 @@ export const EnvironmentPage = connect(stateToProps)(
...
+          <div className="pf-c-form environment-buttons">
             {errorMessage && <Alert isInline className="co-alert" variant="danger" title={errorMessage} />}
             {stale && <Alert isInline className="co-alert" variant="info" title="The information on this page is no longer current.">Click Reload to update and lose edits, or Save Changes to overwrite.</Alert>}
             {success && <Alert isInline className="co-alert" variant="success" title={success} />}
-            {!readOnly &&
-            <button disabled={inProgress} type="submit" className="btn btn-primary" onClick={this.saveChanges}>Save</button>}
-            {!readOnly && <button disabled={inProgress} type="button" className="btn btn-default" onClick={this.reload}>Reload</button>}
+            <ActionGroup>
+              {!readOnly &&
+              <Button disabled={inProgress} type="submit" variant="primary" onClick={this.saveChanges}>Save</Button>}
```
Comment (rhamilto) [inline frontend/public/components/environment.jsx]:
```suggestion
              <Button isDisabled={inProgress} type="submit" variant="primary" onClick={this.saveChanges}>Save</Button>}
```

File: frontend/public/components/environment.jsx [hunk 13]
```diff
@@ -508,13 +508,15 @@ export const EnvironmentPage = connect(stateToProps)(
...
+              {!readOnly && <Button disabled={inProgress} type="button" variant="secondary" onClick={this.reload}>Reload</Button>}
```
Comment (rhamilto) [inline frontend/public/components/environment.jsx]:
```suggestion
              {!readOnly && <Button isDisabled={inProgress} type="button" variant="secondary" onClick={this.reload}>Reload</Button>}
```

File: frontend/__tests__/components/utils/name-value-editor.spec.tsx [hunk 14]
```diff
@@ -49,7 +49,7 @@ describe(NameValueEditor.displayName, () => {
         />
       );
 
-      expect(wrapper.html()).not.toContain('pairs-list__add-icon');
+      expect(wrapper.html()).not.toContain('annotations__add-icon');
```
Comment (rhamilto) [inline frontend/__tests__/components/utils/name-value-editor.spec.tsx]: I prefer the existing name.  This isn't an annotation.  @spadgett?

Comment (sg00dwin) [reply frontend/__tests__/components/utils/name-value-editor.spec.tsx]: reverted to existing name

File: frontend/public/components/operator-lifecycle-manager/create-operand.tsx (hunk 4 above)
Comment (spadgett) [reply frontend/public/components/operator-lifecycle-manager/create-operand.tsx]: > `id="cancel"` is new. Do you need it? If so, it should go on its own line.

I'd rather not add `id` unless necessary (e.g. for associating labels with inputs). If we need it for tests, we should prefer `data-test-id` attributes.

File: frontend/__tests__/components/utils/name-value-editor.spec.tsx (hunk 14 above)
Comment (spadgett) [reply frontend/__tests__/components/utils/name-value-editor.spec.tsx]: > I prefer the existing name. This isn't an annotation. @spadgett?

+1, thanks

File: frontend/public/components/utils/name-value-editor.jsx [hunk 15]
```diff
@@ -78,16 +80,24 @@ export const NameValueEditor = withDragDropContext(class NameValueEditor extends
             readOnly ?
               null :
               <div className="co-toolbar__group co-toolbar__group--left">
-                <button type="button" className="btn btn-link pairs-list__add-btn" onClick={this._append}>
-                  <i aria-hidden="true" className="fa fa-plus-circle pairs-list__add-icon" />{addString}
-                </button>
+                <Button
+                  className="pf-m-link--align-left"
+                  data-test-id="annotations__add-btn"
```
Comment (rhamilto) [inline frontend/public/components/utils/name-value-editor.jsx]:
```suggestion
                  data-test-id="pairs-list__add-btn"
```

File: frontend/integration-tests/views/modal-annotations.view.ts [hunk 16]
```diff
@@ -2,7 +2,7 @@ import { $, $$, browser, ExpectedConditions as until } from 'protractor';
 
 const BROWSER_TIMEOUT = 15000;
 
-const addMoreBtn = $('.pairs-list__add-btn');
+const addMoreBtn = $('[data-test-id="annotations__add-btn"]');
```
Comment (rhamilto) [inline frontend/integration-tests/views/modal-annotations.view.ts]:
```suggestion
const addMoreBtn = $('[data-test-id="pairs-list__add-btn"]');
```

**Comment by spadgett** [comment]
@christianvogt @vojtechszocs @jtomasek FYI, we're working on updating to PF4 buttons

**Comment by rhamilto** [comment]
@sg00dwin, you've got broken unit tests that need fixin'.

**Comment by sg00dwin** [comment]
@rhamilto updates pushed

File: frontend/public/components/environment.jsx [hunk 17]
```diff
@@ -508,13 +508,15 @@ export const EnvironmentPage = connect(stateToProps)(
...
-          <div className="environment-buttons">
+          <div className="pf-c-form environment-buttons">
             {errorMessage && <Alert isInline className="co-alert" variant="danger" title={errorMessage} />}
             {stale && <Alert isInline className="co-alert" variant="info" title="The information on this page is no longer current.">Click Reload to update and lose edits, or Save Changes to overwrite.</Alert>}
             {success && <Alert isInline className="co-alert" variant="success" title={success} />}
-            {!readOnly &&
-            <button disabled={inProgress} type="submit" className="btn btn-primary" onClick={this.saveChanges}>Save</button>}
-            {!readOnly && <button disabled={inProgress} type="button" className="btn btn-default" onClick={this.reload}>Reload</button>}
+            <ActionGroup>
+              {!readOnly &&
```
Comment (rhamilto) [inline frontend/public/components/environment.jsx]: Nit:  move `!readOnly &&` to `<ActionGroup>` and remove from the buttons so you don't end up with an empty `<ActionGroup>`

File: frontend/public/components/monitoring.tsx [hunk 18]
```diff
@@ -1,13 +1,15 @@
...
 import { murmur3 } from 'murmurhash-js';
-import { Alert } from '@patternfly/react-core';
+import { Alert , ActionGroup, Button } from '@patternfly/react-core';
 import { sortable } from '@patternfly/react-table';
 import * as React from 'react';
 import { Helmet } from 'react-helmet';
 import { connect } from 'react-redux';
 import { Link, Redirect, Route, Switch } from 'react-router-dom';
 
+import { PlusCircleIcon } from '@patternfly/react-icons';
```
Comment (rhamilto) [inline frontend/public/components/monitoring.tsx]: Nit:  remove empty line above.

**Pull request merged**